### PR TITLE
Redis Integration

### PR DESCRIPTION
**Pull request by prashantpawar**
Working and tested endpoints for your consumption.

Attaching Postman's profile (install [Postman](https://chrome.google.com/webstore/detail/postman/fhbjgbiflinjbdggehcddcbncdddomop?hl=en) the API client, and import the following profile into it, to be able to test the endpoints for candidate service.

[CandidateService.json.postman_collection.zip](https://github.com/inrhythm/vivace/files/65766/CandidateService.json.postman_collection.zip)

Also check out README.md to see the instructions on running Redis server for development purposes.

### CODE REVIEW

**Comment by jimitpatel9** [comment]
:+1:

File: test/server/server.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import proxyquire from 'proxyquire';
+//import app from '../../server/server';
+import { should } from 'chai';
+
+
+describe('server', () => {
+  it('should serve index.html', () => {
+    "10".should.be.equal("10");
```
Comment (jackofseattle) [inline test/server/server.js]: weird test

**Comment by jackofseattle** [comment]
LGTM

File: test/server/server.js (hunk 1 above)
Comment (prashantpawar) [inline test/server/server.js]: Lol mistaken commit.

**Comment by jshor** [comment]
records need a primary key (as discussed)

File: server/helper/database.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import flat from 'flat';
+import process from 'process';
+import bluebird from 'bluebird';
+
+class DbHelper {
+
+  constructor() {
+    //Promisifying the redis library
+    bluebird.promisifyAll(redis.RedisClient.prototype);
+    bluebird.promisifyAll(redis.Multi.prototype);
```
Comment (jshor) [inline server/helper/database.js]: I thought promises were native in node now

File: server/server.js [hunk 3]
```diff
@@ -32,3 +35,8 @@ app.get('/', (req, res) => {
 let listener = app.listen(8001, () => {
   console.log('\x1b[33m%s:\x1b[4m%s\x1b[0m', 'App is listening on port', listener.address().port);
```
Comment (jshor) [inline server/server.js]: wut

**Comment by jshor** [comment]
The following appears in the array when performing a `GET` on `/api/candidates`

```

  {
    "code": "WRONGTYPE",
    "command": "HGETALL"
  },
```

File: server/helper/database.js (hunk 2 above)
Comment (prashantpawar) [inline server/helper/database.js]: I am using [promisify](http://bluebirdjs.com/docs/api/promise.promisify.html) feature of bluebird.

File: server/server.js (hunk 3 above)
Comment (prashantpawar) [inline server/server.js]: Lol, I didn't wanna include a third party library to display colors, but that's how you display colors on console.

File: server/helper/database.js (hunk 2 above)
Comment (jshor) [inline server/helper/database.js]: Ah, gotcha. I was thinking `promisifyAll()` was built in to `Promise`

**Comment by jshor** [comment]
LGTM

**Pull request merged**